include/tvm/runtime/container.h

```diff
@@ -956,6 +956,19 @@ class Array : public ObjectRef {
     return static_cast<ArrayNode*>(data_.get());
   }
 
+  /*!
+   * \brief Concat with another Array. It does not mutate the current one.
+   * \param other Array to be concatenated.
+   * @return The concatenated Array. Original Array is kept unchanged.
+   */
+  Array<T> Concat(const Array<T>& other) const {
```

Comment:
Consider make it as a global function, which also enables copy on write on the lhs(this)